internal/controller/clustersecurityintentbinding_controller.go

```diff
@@ -228,15 +267,292 @@ func (r *ClusterSecurityIntentBindingReconciler) findCsibsForSi(ctx context.Cont
 	return requests
 }
 
-func (r *ClusterSecurityIntentBindingReconciler) updateCsibStatus(ctx context.Context, logger logr.Logger, req ctrl.Request) error {
+type npTrackingObj struct {
+	create bool
+	update bool
+	np     *v1alpha1.NimbusPolicy
+}
+
+type nsTrackingObj struct {
+	ns *corev1.Namespace
+}
+
+// we should not create object in these ns
+var nsBlackList = []string{"kube-system"}
+
+const wildcard = "*"
+
+func (r *ClusterSecurityIntentBindingReconciler) isValidCsib(ctx context.Context, logger logr.Logger, req ctrl.Request) bool {
+
+	// get the csib
+	var csib v1alpha1.ClusterSecurityIntentBinding
+	if err := r.Get(ctx, req.NamespacedName, &csib); err != nil {
+		logger.Error(err, "failed to fetch ClusterSecurityIntentBinding", "ClusterSecurityIntentBinding.Name", req.Name)
+		return false
+	}
+
+	// validate the CSIB.
+	excludeLen := len(csib.Spec.Selector.NsSelector.ExcludeNames)
+	matchLen := len(csib.Spec.Selector.NsSelector.MatchNames)
+	if matchLen > 0 && excludeLen > 0 {
+		err := errors.New("invalid clustersecurityintentbinding")
+		logger.Error(err, "Both MatchNames and ExcludeNames should not be set", "ClusterSecurityIntentBinding.Name", req.Name)
+		return false
+	}
+	if matchLen == 0 && excludeLen == 0 {
+		err := errors.New("invalid clustersecurityintentbinding")
+		logger.Error(err, "Atleast one of MatchNames or ExcludeNames should be set", "ClusterSecurityIntentBinding.Name", req.Name)
+		return false
+	}
+	// In MatchNames, if a  "*" is present, it should be the only entry
+	for i, ns := range csib.Spec.Selector.NsSelector.MatchNames {
+		if ns == wildcard && i > 0 {
+			err := errors.New("invalid clustersecurityintentbinding")
+			logger.Error(err, "If * is present, it should be only entry", "ClusterSecurityIntentBinding.Name", req.Name)
+			return false
+		}
+	}
+
+	return true
+}
+
+func (r *ClusterSecurityIntentBindingReconciler) createOrUpdateNp(ctx context.Context, logger logr.Logger, req ctrl.Request) error {
```

Comment:
This method could benefit from being split up.  Remember the general rule of thumb: functions/methods are more maintainable when they are kept concise.

Reply:
I agree with the overall principle about maintainability, but do not see benefit of splitting up this function in multiple functions.